### DISCUSSION TITLE
socketio server setup issue for python versions >3.7.5

### DISCUSSION THREAD

**Discussion by David-Vithayathil** [miguelgrinberg/python-socketio#864] [Q&A]
We were setting up socketio server using python 3.7.5 and the js client seems to communicate properly.
On switching to python3.8, the server doesn't setup with asyncio using aiohttp as per [this example](https://python-socketio.readthedocs.io/en/latest/intro.html#:~:text=from%20aiohttp%20import%20web) (No issue when using eventlet example). 

I have attached the code used to setup the server using python 3.8

```python
import asyncio
import socketio
from aiohttp import web

sio = socketio.AsyncServer(async_mode='aiohttp', async_handlers=True, cors_allowed_origins='*')
app = web.Application()
sio.attach(app)

@sio.on('connect')
async def connect(sid, environ):
    print("connected: ", sid)

@sio.on('disconnect')
async def disconnect(sid):
    print('disconnect ', sid)

if __name__ == '__main__':
    web.run_app(app, host='localhost', port=65000)
```

We switched between python 3.7.5 and 3.8 which uses the same requirements and script, the issue was reproducible only with python3.8.

Here is the js client

```javascript

const socket = io('http://localhost:65000', { 
        reconnection: true,
        rejectUnauthorized: false,
        transports: ['websocket']
    })
socket.on("connect_error", (err) => {
            console.log(err.message);
        });
socket.on("connect", () => {
    console.log(socket.connected); // true
});
socket.connect()
console.log("CONNECTION STATUS : ", socket.connected);
```

I checked the [tox file](https://github.com/miguelgrinberg/python-socketio/blob/main/tox.ini#L2) and it does support python3.8, I assume that the [unit tests](https://github.com/miguelgrinberg/python-socketio/blob/main/tests/asyncio/test_asyncio_server.py) do pass for this version, which is why I didn't raise this as an issue.

Here is the pip freeze
```
aioredis==1.3.1
aiohttp==3.5.4
eventlet==0.33.0
python-socketio==5.4.1
python-engineio==4.3.0
websockets==9.1
websocket-client==0.56.0
```
javascript client is using version "^4.3.2"

**Comment by miguelgrinberg** [comment]
You haven't indicated what the error or problem is when using 3.8. Can you please expand on that?

**Reply by David-Vithayathil** [reply]
There are no errors thrown from the server script, it hangs after printing this

```
======== Running on http://0.0.0.0:65000 ========
```

Ideally it should print the text 'connect' along with the sid as per the script during the connect event, like this
```
======== Running on http://0.0.0.0:65000 ========
(Press CTRL+C to quit)
connected:  TdQkBBVUmbL3jTIHAAAB
```
This works as expected when I use python3.7.5 but not with 3.8.0.

The js client only throws websocket error during the connect_error event. That happened when the server script was not run and keeps throwing the error after the server script was run.

Again this happens only for the setup with asyncio using aiohttp and there is no issue when I use the eventlet example.

**Reply by miguelgrinberg** [reply]
I'm running your server here, with python 3.8, and I don't see this. The `Press Ctrl-C` message appears just fine, and I can make connections.

**Reply by David-Vithayathil** [reply]
The issue lies with the JavaScript client (Although the [version compatibility](https://python-socketio.readthedocs.io/en/latest/intro.html#version-compatibility) does match) . I **should** have tested out with python client first before opening the discussion.

Since there is no issue with socketio python client and server communication, I am closing this discussion here.
